### PR TITLE
[search] Simplifications of Retrieval. Removed code duplication.

### PR DESCRIPTION
**Pull request by ygorshenin**
@vng or @mpimenov PTAL

### CODE REVIEW

File: search/retrieval.cpp [hunk 1]
```diff
@@ -78,13 +78,26 @@ class EditedFeaturesHolder
...
   {
-    for_each(m_modified.begin(), m_modified.end(), fn);
-    for_each(m_created.begin(), m_created.end(), fn);
+    ReadFeatures(m_modified, fn);
+    ReadFeatures(m_created, fn);
   }
 
 private:
+  template <typename TFn>
+  void ReadFeatures(vector<uint32_t> const & features, TFn & fn)
```
Comment (mpimenov) [inline search/retrieval.cpp]: s/ReadFeatures/ForEach/ ?

Comment (ygorshenin) [inline search/retrieval.cpp]: Done

**Comment by ygorshenin** [comment]
@mpimenov PTAL

**Comment by mpimenov** [comment]
LGTM

**Comment by ygorshenin** [comment]
Thanks for the review!

**Pull request merged**